src/payment/strategies/adyenv2/adyenv2.ts

```diff
@@ -158,16 +162,11 @@ export interface InputDetail {
     value?: string;
 }
 
-export interface AdyenThreeDS2Options {
-    /**
-     * Specify Three3DS2Challenge Widget Size
-     */
-    widgetSize?: string;
-
+export interface AdditionalActionCallbacks {
     /**
-     * A callback that gets called when adyen component is mounted
+     * A callback that gets called when adyen component is loaded
      */
-    onLoad(cancel: () => void): void;
+    onLoad(cancel: () => void, shouldShow: boolean): void;
```

Comment:
Based on this `shouldShow` flag, it seems to me that the requirements for 3DS and Bank Transfer are quite different, but you're trying to meet them with a single solution. IMO, they should be tackled separately.

3DS requires us to present a challenge to the shopper, which they must complete in order to proceed. So it requires two things on our end - a visible container to insert the challenge widget into, and a way for us to dismiss or unmount the challenge widget if the shopper chooses not to continue.

On the other hand, Bank Transfer doesn't require us to present any UI component to the shopper when the initial payment fails. Instead, it just needs to redirect the shopper elsewhere. Therefore it doesn't really need a visible container, and it doesn't have anything to unmount.

Having `shouldShow` flag is an indicator that the client needs to treat the two scenarios differently. I'm guessing you need this flag because in UCO we're presenting the 3DS challenge inside a modal, which gets triggered inside the `onLoad` callback. As you're using the same `onLoad` callback for Bank Transfer, the modal for 3DS will also be triggered for Bank Transfer flow unless it is manually suppressed. It is _a_ solution - but it is very specific to how we (UCO) present the 3DS challenge to shoppers. Imagine there's a custom checkout that decides to present the 3DS challenge inline inside of a modal; this `shouldShow` flag won't have any use.

So it seems to me that the most appropriate solution is to not to introduce a new `additionalActionOptions` and not deprecate `threeDS2Options`. Instead, you should treat 3DS and Bank Transfer separately. Correct me if I'm wrong, but it seems for Bank Transfer, the "component" that gets returned from `createFromAction` is not a visible one. Instead it is a hidden component whose responsibility is to redirect shoppers. If that's the case, then there's actually no need for the client to provide a separate container ID and a callback so you can use it to unmount it. You can probably just create new hidden container behind the scene and just mount it there.